pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -365,9 +360,17 @@ func (h *AutoscalersController) deleteAutoscaler(obj interface{}) {
 
 	deletedHPA, ok := obj.(*autoscalingv2.HorizontalPodAutoscaler)
 	if ok {
-		log.Debugf("Deleting Metrics from HPA %s/%s", deletedHPA.Namespace, deletedHPA.Name)
 		toDelete := hpa.Inspect(deletedHPA)
-		h.store.DeleteExternalMetricValues(toDelete)
+		h.deleteFromLocalStore(toDelete)
+		log.Debugf("Deleting %s/%s from the local cache", deletedHPA.Namespace, deletedHPA.Name)
+		if !h.le.IsLeader() {
+			return
+		}
+		log.Infof("Deleting entries of metrics from HPA %s/%s in the Global Store", deletedHPA.Namespace, deletedHPA.Name)
+		if err := h.store.DeleteExternalMetricValues(toDelete); err != nil {
+			h.queue.AddRateLimited(deletedHPA)
```

Comment:
`h.queue.AddRateLimited()` should take as argument the object `key` and not the object himself

Reply:
same the `h.queue.Done()` line 374

Reply:
in fact you already have a fonction to enqueue and object properly
`func (h *AutoscalersController) enqueue(obj interface{})` line 401

Reply:
Good catch. That one had been bothering me, thanks for the details